tests/forte/processors/data_augment/algorithms/eda_processors_test.py

```diff
@@ -42,124 +43,72 @@ def setUp(self):
-                self.assertEqual(token.text, expected_tokens[j])
+            for j, token in enumerate(aug_pack.get(Token)):
+                self.assertEqual(token.text, expected_tokens[idx][j])
 
-    @data((["Mary and Samantha arrived at the bus station early but waited until noon for the bus."],))
+    @data(
+        (["Mary and Samantha arrived at the bus station early but waited until noon for the bus."],
+         ["await Mary and Samantha arrived at the bus station early but waited until noon for the bus."],
+         [['await ', 'Mary', 'and', 'Samantha', 'arrived', 'at', 'the', 'bus', 'station', 'early', 'but', 'waited', 'until', 'noon', 'for', 'the', 'bus', '.']],
+        )
+    )
     @unpack
-    def test_random_insert(self, texts):
-        for idx, text in enumerate(texts):
-            file_path = os.path.join(self.test_dir, f"{idx + 1}.txt")
-            with open(file_path, 'w') as f:
-                f.write(text)
-
-        expected_text = \
-            "await Mary and Samantha arrived at the bus station early but waited until noon for the bus.\n"
-        expected_tokens = ['await ', 'Mary', 'and', 'Samantha', 'arrived', 'at', 'the', 'bus', 'station', 'early',
-                           'but', 'waited', 'until', 'noon', 'for', 'the', 'bus', '.']
+    def test_random_insert(self, texts, expected_outputs, expected_tokens):
+        self.nlp.add(component=RandomInsertionDataAugmentProcessor())
+        self.nlp.initialize()
 
-        insert_processor = RandomInsertionDataAugmentProcessor()
-        insert_processor.initialize(resources=None, configs=insert_processor.default_configs())
+        for idx, m_pack in enumerate(self.nlp.process_dataset(texts)):
+            aug_pack = m_pack.get_pack('augmented_input_src')
 
-        for idx, m_pack in enumerate(self.nlp.process_dataset(self.test_dir)):
-            src_pack = m_pack.get_pack('input_src')
-            tgt_pack = m_pack.get_pack('output_tgt')
+            self.assertEqual(aug_pack.text, expected_outputs[idx])
 
-            # Copy the source pack to target pack.
-            tgt_pack.set_text(src_pack.text)
-            for anno in src_pack.get(Annotation):
-                new_anno = type(anno)(
-                    tgt_pack, anno.begin, anno.end
-                )
-                tgt_pack.add_entry(new_anno)
+            for j, token in enumerate(aug_pack.get(Token)):
+                self.assertEqual(token.text, expected_tokens[idx][j])
 
-            insert_processor._process(m_pack)
+    @data(
+        (["Mary and Samantha arrived at the bus station early but waited until noon for the bus."],
+         ["Mary and   at  bus   but waited until  for the ."],
+         [['Mary', 'and', 'at', 'bus', 'but', 'waited', 'until', 'for', 'the', '.']],
+        )
+    )
+    @unpack
+    def test_random_delete(self, texts, expected_outputs, expected_tokens):
+        self.nlp.add(component=RandomDeletionDataAugmentProcessor(), config={"alpha": 0.5})
```

Comment:
I am curious (and not sure). Will `self.nlp` contains the other data augment processor since self.nlp is shared? I am not fully confident about how unittest works.

Reply:
I believe `setUp` is called before each test is conducted.

Reply:
ok thanks